### PR TITLE
fix(query): input format TSV  should not use quote for enclosing

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

not work for clickhouse httphandler: https://github.com/datafuselabs/databend/issues/6669

Fixes  https://github.com/datafuselabs/databend/issues/6554
https://github.com/datafuselabs/databend/issues/6553
https://github.com/datafuselabs/databend/issues/6650
https://github.com/datafuselabs/databend/issues/6650

### CODE REVIEW

File: tests/suites/1_stateful/00_copy_v2/00_0000_copy_tsv.result [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---1
+"a	1	2
+"a"	2	3
+"a	3	2
+'a	4	2
+'a'	5	3
+'a	6	2
+'a"b'	7	5
+\\	8	1
+\\i	9	2
```
Comment (Xuanwo) [inline tests/suites/1_stateful/00_copy_v2/00_0000_copy_tsv.result:10]: Add a case that contains an empty space ` `? :

```
'a" b' 7 5
```

**Review by Xuanwo — APPROVED** [review]
Mostly LGTM, please make Clippy happy~

File: common/io/src/buffer/buffer_read_ext.rs [hunk 2]
```diff
@@ -122,11 +122,20 @@ pub trait BufferReadExt: BufferRead {
...
-            // TODO parse complex escape sequence
-            buf.push(b'\\');
-            self.read_escaped_string_text(buf)?;
+        loop {
+            self.keep_read(buf, |f| f != b'\t' && f != b'\n' && f != b'\\')?;
+            if self.ignore_byte(b'\\')? {
+                let buffer = self.fill_buf()?;
+                let c = buffer[0];
+                match c {
+                    b'\'' | b'\"' | b'\\' => {}
```
Comment (sundy-li) [inline common/io/src/buffer/buffer_read_ext.rs:131]: Some other cases in [clickhouse](https://github.com/ClickHouse/ClickHouse/blob/aa737b7a11167fa4af3bb8f8dc4e70909d1728e2/src/IO/ReadHelpers.cpp):


1. `\x2A` to read hex codes
2. `\N`  empty string, skipped
3. ...

**Comment by youngsofun** [comment]
@Xuanwo  only copy from ontime_200.csv.zst fail,  is there anything special about zstd?

File: common/io/src/buffer/buffer_read_ext.rs (hunk 2 above)
Comment (youngsofun) [reply common/io/src/buffer/buffer_read_ext.rs:131]: ok,  these are not as important (at least not fail the loading),
I will add them after the tests passed

**Comment by youngsofun** [comment]
log error like this 

tests/suites/1_stateful/00_copy_v2/00_0000_copy_from_stage2.py


```
 1 | CREATE TABLE ontime200
   | ------ while parsing `CREATE TABLE [IF NOT EXISTS] [<database>.]<table> [<source>] [<table_options>]`
   .
35 |     DepDelayMinutes                 INT,
36 |     DepDel15                        INT,
37 |     DepartureDelayGroups            VARCHAR,
38 |     DepTimeBlk                      V    DepTimeBlk                              INT,
   |     ----------                      ^
   |     |                               |
   |     |                               expected `BOOLEAN`, `BOOL`, `UINT8`, `TINYINT`, `UINT16`, `SMALLINT`, or 25 more ...
   |     |                               while parsing type name
   |     while parsing `<column name> <type> [NOT NULL | NULL] [DEFAULT <default value>] [COMMENT '<comment>']`


````

**Comment by BohuTANG** [comment]
```
Diff in /workspace/query/src/pipelines/processors/sources/file_splitter.rs at line 99:
         rows_to_skip: &mut u64,
     ) -> Result<()> {
         if data.is_empty() {
-            return Ok(())
+            return Ok(());
         }
         let mut data_slice = data;
Error: Process completed with exit code 1.
```

File: common/io/src/buffer/buffer_read_ext.rs (hunk 2 above)
Comment (youngsofun) [reply common/io/src/buffer/buffer_read_ext.rs]: done

**Review by sundy-li — APPROVED** [review]

**Pull request merged**